From 6cfdc840159bb349310c3b81cd2df949f1522760 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Wed, 3 Jul 2024 16:01:08 +0200
Subject: [PATCH] frdp-session: Turn off asynchronous connection

Turn off the asynchronous connection via GTask thread as the FreeRDP
process runs in the new thread which makes it hard to cooperate with
the original thread running UI leading to random race conditions.
Turn it on again if there will be an async support for connection
added to FreeRDP.

The disadvantage is that the application freezes during connection
for some time.
---
 src/frdp-session.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/src/frdp-session.c b/src/frdp-session.c
index 1bb5536..fd8bc46 100644
--- a/src/frdp-session.c
+++ b/src/frdp-session.c
@@ -1315,7 +1315,19 @@ frdp_session_connect (FrdpSession         *self,
   self->priv->port = port;
 
   task = g_task_new (self, cancellable, callback, user_data);
-  g_task_run_in_thread (task, frdp_session_connect_thread);
+  /* Turn off the asynchronous connection via GTask thread as the FreeRDP
+   * process then runs in the new thread which makes it hard to cooperate
+   * with the original thread running UI leading to random race conditions.
+   * Turn it on again if there will be an async support for connection
+   * added to FreeRDP.
+   * The disadvantage is that the application freezes during connection
+   * for some time.
+
+     g_task_run_in_thread (task, frdp_session_connect_thread);
+
+   */
+
+  frdp_session_connect_thread (task, self, user_data, cancellable);
 
   g_object_unref (task);
 }
-- 
2.47.1

